plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java

```diff
@@ -822,6 +822,74 @@ public void testLargeDefaultDomainCompactionThreshold()
                 "VALUES('" + propertyName + "','1000', '1000', 'integer', 'Maximum ranges to allow in a tuple domain without simplifying it')");
     }
 
+    @Test
+    public void testFloatPredicatePushdown()
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_float_predicate_pushdown",
+                """
+                        (
+                        c_real real,
+                        c_real_neg_infinity real,
+                        c_real_pos_infinity real,
+                        c_real_nan real,
+                        c_double double,
+                        c_double_neg_infinity double,
+                        c_double_pos_infinity double,
+                        c_double_nan double)""",
+                List.of("3.14, -infinity(), +infinity(), nan(), 3.14, -infinity(), +infinity(), nan()"))) {
+            assertThat(query("""
+                    SELECT c_real, c_double, c_double_neg_infinity, c_double_pos_infinity, c_double_nan FROM %s
+                    WHERE c_real = real '3.14'""".formatted(table.getName())))
+                    .matches("VALUES (real '3.14', double '3.14', -infinity(), +infinity(), nan())")
```

Comment:
Why do we want to check values of `c_real, c_double, c_double_neg_infinity, c_double_pos_infinity, c_double_nan`? That looks unrelated for verifying predicate pushdown. How about adding another column (e.g.`id int`)?

Reply:
I'll just reuse the existing one `c_real`